src/components/CSVButton/CSVButton.jsx

```diff
@@ -0,0 +1,258 @@
+        'Pre-Test A',
+        'Pre-Test A Notes',
+        'Pre-Test R',
+        'Pre-Test R Notes',
+      ];
+    }
+    if (type === 'site') {
+      return [
+        'Site ID',
+        'Student ID',
+        'First Name',
+        'Last Name',
+        'Grade',
+        'Home Teacher',
+        'Gender',
+        'Ethnicity',
+        'Student Group ID',
+        'Post-Test A',
+        'Post-Test A Notes',
+        'Post-Test R',
+        'Post-Test R Notes',
+        'Pre-Test A',
+        'Pre-Test A Notes',
+        'Pre-Test R',
+        'Pre-Test R Notes',
+      ];
+    }
+    return [];
+  }
+
+  function mapResponse() {
+    if (type === 'allAreas') {
+      return studentResponseData.map(student => [
+        student.areaId,
+        student.studentId,
+        student.firstName,
+        student.lastName,
+        student.grade,
+        student.homeTeacher,
+        student.gender,
+        student.ethnicity,
+        student.studentGroupId,
+        student.posttestA,
+        student.posttestANotes,
+        student.posttestR,
+        student.posttestRNotes,
+        student.pretestA,
+        student.pretestANotes,
+        student.pretestR,
+        student.pretestRNotes,
+      ]);
+    }
+    if (type === 'area') {
+      return studentResponseData.map(student => [
+        student.areaId,
+        student.studentId,
+        student.firstName,
+        student.lastName,
+        student.grade,
+        student.homeTeacher,
+        student.gender,
+        student.ethnicity,
+        student.studentGroupId,
+        student.posttestA,
+        student.posttestANotes,
+        student.posttestR,
+        student.posttestRNotes,
+        student.pretestA,
+        student.pretestANotes,
+        student.pretestR,
+        student.pretestRNotes,
+      ]);
+    }
+    if (type === 'site') {
+      return studentResponseData.map(student => [
+        student.siteId,
+        student.studentId,
+        student.firstName,
+        student.lastName,
+        student.grade,
+        student.homeTeacher,
+        student.gender,
+        student.ethnicity,
+        student.studentGroupId,
+        student.posttestA,
+        student.posttestANotes,
+        student.posttestR,
+        student.posttestRNotes,
+        student.pretestA,
+        student.pretestANotes,
+        student.pretestR,
+        student.pretestRNotes,
+      ]);
+    }
+    return [];
+  }
+
+  const CSVReport = {
+    data: studentResponseData,
+    headers: createHeaders(),
+    filename: 'Areas_Report.csv',
+  };
+
+  const addAssociatedSiteToArea = async resData => {
+    async function fetchStudentsByAllAreas() {
+      // eslint-disable-next-line no-restricted-syntax
+      for (const element of resData) {
+        TLPBackend.get(`/students/area/${element.areaId}`)
+          .then(res => {
+            res.data.forEach(student => {
+              // eslint-disable-next-line no-param-reassign
+              student.areaId = element.areaId;
```

Comment:
Also a bit confused by this - the student's areaId is already given in the response. Why do we need to reassign it?

Reply:
<img width="560" alt="image" src="https://user-images.githubusercontent.com/48849590/167222267-df2bde4f-13e9-4fdf-aa95-724a4b41a8f5.png">

I don't see the areaId in the student response before this assignment. If it was in the student response, I agree, there wouldn't need to loop through it so much